src/SaveHandler/DbTableGateway.php

```diff
@@ -149,10 +149,14 @@ public function write($id, $data)
      */
     public function destroy($id)
     {
-        return (bool) $this->tableGateway->delete([
+        $exists = (bool) $this->read($id);
+
+        $deleted = (bool) $this->tableGateway->delete([
             $this->options->getIdColumn()   => $id,
             $this->options->getNameColumn() => $this->sessionName,
         ]);
+
+        return $exists ? $deleted : true;
```

Comment:
I've tested this locally, and we can indeed return early. I'm making that change as part of the merge.